core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -805,7 +806,7 @@ class DAGScheduler(
     }
 
     val properties = if (jobIdToActiveJob.contains(jobId)) {
-      jobIdToActiveJob(stage.jobId).properties
+      jobIdToActiveJob(stage.jobId).properties.orNull
```

Comment:
I don't know if there's a good reason for this, but I don't think we can change it at this point without breaking binary compatibility.  We could use annotations / comments to make those fields' nullability more apparent, though.